exchanges/poloniex/poloniex_futures.go

```diff
@@ -0,0 +1,769 @@
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	var resp []*FuturesPosition
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, fGetPositionHistoryEPL, http.MethodGet, tradePathV3+"position/history", params, nil, &resp)
+}
+
+// AdjustMarginForIsolatedMarginTradingPositions add or reduce margin for positions in isolated margin mode.
+func (e *Exchange) AdjustMarginForIsolatedMarginTradingPositions(ctx context.Context, symbol, positionSide, adjustType string, amount float64) (*AdjustFuturesMarginResponse, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	if amount <= 0 {
+		return nil, limits.ErrAmountBelowMin
+	}
+	if adjustType == "" {
+		return nil, errMarginAdjustTypeMissing
+	}
+	arg := &struct {
+		Symbol       string  `json:"symbol"`
+		PositionSide string  `json:"posSide,omitempty"`
+		Amount       float64 `json:"amt,string"`
+		Type         string  `json:"type"`
+	}{
+		Symbol:       symbol,
+		PositionSide: positionSide,
+		Amount:       amount,
+		Type:         adjustType,
+	}
+	var resp *AdjustFuturesMarginResponse
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, fAdjustMarginEPL, http.MethodPost, tradePathV3+"position/margin", nil, arg, &resp)
+}
+
+// GetFuturesLeverage retrieves the list of leverage.
+func (e *Exchange) GetFuturesLeverage(ctx context.Context, symbol, marginMode string) ([]*FuturesLeverage, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	if marginMode != "" {
+		params.Set("mgnMode", marginMode)
+	}
+	var resp []*FuturesLeverage
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, fGetPositionLeverageEPL, http.MethodGet, positionPathV3+"leverages", params, nil, &resp)
+}
+
+// SetFuturesLeverage change leverage
+func (e *Exchange) SetFuturesLeverage(ctx context.Context, symbol, marginMode, positionSide string, leverage int64) (*FuturesLeverage, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	if marginMode == "" {
+		return nil, margin.ErrInvalidMarginType
+	}
+	if positionSide == "" {
+		return nil, order.ErrSideIsInvalid
+	}
+	if leverage <= 0 {
+		return nil, order.ErrSubmitLeverageNotSupported
+	}
+	var resp *FuturesLeverage
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, fSetPositionLeverageEPL, http.MethodPost, positionPathV3+"leverage", nil, &map[string]string{
+		"symbol":  symbol,
+		"mgnMode": marginMode,
+		"posSide": positionSide,
+		"lever":   strconv.FormatInt(leverage, 10),
+	}, &resp)
+}
+
+// SwitchPositionMode switch the current position mode. Please ensure you do not have open positions and open orders under this mode before the switch.
+// Position mode, HEDGE: LONG/SHORT, ONE_WAY: BOTH
+func (e *Exchange) SwitchPositionMode(ctx context.Context, positionMode string) error {
+	if positionMode == "" {
```

Comment:
🚧 Change request:
Can we enforce this please?
```suggestion
	if positionMode != "HEDGE" && positionMode != "ONE_WAY" {
```